docs/06-ReusingTestCode.md

```diff
@@ -0,0 +1,636 @@
+
+But let's move further. A PageObject concept also defines that methods for the page interaction should also be stored in a PageObject class. It now stores passed instance of Actor class. An AcceptanceTester can be accessed via `AcceptanceTester` property of that class. Let's define a `login` method in this class.
+
+```php
+<?php
+class UserLoginPage
+{
+    // include url of current page
+    public static $URL = '/login';
+
+    /**
+     * @var AcceptanceTester
+     */
+    protected $tester;
+
+    public function __construct(AcceptanceTester $I)
+    {
+        $this->tester = $I;
+    }
+
+    public function login($name, $password)
+    {
+        $I = $this->tester;
+
+        $I->amOnPage(self::$URL);
+        $I->fillField(LoginPage::$usernameField, $name);
+        $I->fillField(LoginPage::$passwordField, $password);
+        $I->click(LoginPage::$loginButton);
+
+        return $this;
+    }    
+}
+?>
+```
+
+And here is an example of how this PageObject can be used in a test.
+
+```php
+<?php
+$I = new AcceptanceTester($scenario);
+$loginPage = new \Page\Login($I);
+$loginPage->login('bill evans', 'debby');
+$I->amOnPage('/profile');
+$I->see('Bill Evans Profile', 'h1');
+?>
+```
+
+If you write your scenario-driven tests in Cest format (which is recommended approach), you can bypass manual creation of PageObject and delegate this task to Codeception. If you specify which objects you need for a test, Codeception will try to create it using it's dependency injection container. In case of PageObject you should declare a class as a parameter for a test method:
+
+```php
+<?php
+class UserCest 
+{    
+    function showUserProfile(AcceptanceTester $I, \Page\Login $loginPage)
+    {
+        $loginPage->login('bill evans', 'debby');
+        $I->amOnPage('/profile');
+        $I->see('Bill Evans Profile', 'h1');        
+    }
+}
+?>
+```
+
+Dependency Injection container can construct any object that require any known class type. For instance, `Page\Login` required `AcceptanceTester`, and so it was injected into `Page\Login` constructor, and PageObject was created and passed into method arguments. You should specify explicitly types of requried objects for Codeception to know what objects should be created for test. Dependency Injection will be described in next chapter. 
+
+## Modules and Helpers
+
+In examples above we only grouped actions into one. What happens when we need to create a custom action? 
+In this case it's a good idea to define missing actions or assertion commands in custom modules, which are called Helpers. They can be found in the `tests/_support/Helper` directory.
+
+<div class="alert alert-info">
+We already know how to create custom login method in AcceptanceTester class. We used actions from standard modules and combined them to make user easier to log in. Helpers allow us to create **new actions** unrelated to standard modules (or using their internals).  
```

Comment:
know how to create **a** custom login

them to make **it easier for a** user ~~easier~~ to log in